app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def posts_counter_update
+    author.update(posts_counter: author.posts.count)
+  end
```

Comment:
- [ ] [optional] @mryeminthu Great work implementing the change requested to add a method to update the post counter, I noted you didn't use [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to the database, then make the update_post_counter method private, using this will make your method work efficiently updating the court whenever a post is saved in the database as you did for comments and likes.
```suggestion
```ruby
 after_save :posts_counter_update

  private

  def posts_counter_update
    author.update(posts_counter: author.posts.count)
  end
```